azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/azurecli/AzureCliAccount.java

```diff
@@ -31,35 +35,45 @@ public AuthType getAuthType() {
         return Mono.fromCallable(() -> {
             AzureCliUtils.ensureMinimumCliVersion();
             AzureCliUtils.executeAzCommandJson("az account get-access-token --output json");
-            return true;
-        });
-    }
+            List<AzureCliSubscription> subscriptions = AzureCliUtils.listSubscriptions();
+            if (subscriptions.isEmpty()) {
+                throw new AzureToolkitAuthenticationException("Cannot find any subscriptions in current account.");
+            }
 
-    protected Mono<TokenCredentialManager> createTokenCredentialManager() {
-        List<AzureCliSubscription> subscriptions = AzureCliUtils.listSubscriptions();
-        if (subscriptions.isEmpty()) {
-            throw new AzureToolkitAuthenticationException("Cannot find any subscriptions in current account.");
-        }
+            AzureCliSubscription defaultSubscription = subscriptions.stream()
+                    .filter(AzureCliSubscription::isSelected).findFirst().orElse(subscriptions.get(0));
 
-        AzureCliSubscription defaultSubscription = subscriptions.stream()
-                .filter(AzureCliSubscription::isSelected).findFirst().orElse(subscriptions.get(0));
+            AzureEnvironment configEnv = Azure.az(AzureCloud.class).get();
 
-        this.entity.setEmail(defaultSubscription.getEmail());
+            if (configEnv != null && defaultSubscription.getEnvironment() != configEnv) {
+                throw new AzureToolkitAuthenticationException(
+                        String.format("The azure cloud from azure cli '%s' doesn't match with your auth configuration, " +
+                                        "you can change it by executing 'az cloud set --name=%s' command to change the cloud in azure cli.",
+                                AzureEnvironmentUtils.getCloudNameForAzureCli(defaultSubscription.getEnvironment()),
+                                AzureEnvironmentUtils.getCloudNameForAzureCli(configEnv)));
+            }
 
-        subscriptions = subscriptions.stream().filter(s -> StringUtils.equals(this.entity.getEmail(), s.getEmail()))
-                .collect(Collectors.toList());
+            this.entity.setEnvironment(defaultSubscription.getEnvironment());
+            this.entity.setEmail(defaultSubscription.getEmail());
 
-        // use the tenant who has one or more subscriptions
-        this.entity.setTenantIds(subscriptions.stream().map(Subscription::getTenantId).distinct().collect(Collectors.toList()));
+            subscriptions = subscriptions.stream().filter(s -> StringUtils.equals(this.entity.getEmail(), s.getEmail()))
```

Comment:
May need add `env` as filter as `AzureCliUtils.listSubscriptions()` will list all subscriptions in azure cli, including subs in other envinronment